tdp/cli/commands/resume.py

```diff
@@ -0,0 +1,121 @@
+    envvar="TDP_DATABASE_DSN",
+    required=True,
+    type=str,
+    help=(
+        "Database Data Source Name, in sqlalchemy driver form "
+        "example: sqlite:////data/tdp.db or sqlite+pysqlite:////data/tdp.db. "
+        "You might need to install the relevant driver to your installation (such "
+        "as psycopg2 for postgresql)"
+    ),
+)
+@click.option(
+    "--collection-path",
+    "collections",
+    envvar="TDP_COLLECTION_PATH",
+    required=True,
+    callback=collection_paths_to_collections,  # transforms into Collections object
+    help=f"List of paths separated by your os' path separator ({os.pathsep})",
+)
+@click.option(
+    "--run-directory",
+    envvar="TDP_RUN_DIRECTORY",
+    type=Path,
+    help="Working directory where the executor is launched (`ansible-playbook` for Ansible)",
+    required=True,
+)
+@click.option(
+    "--vars",
+    envvar="TDP_VARS",
+    required=True,
+    type=click.Path(resolve_path=True, path_type=Path),
+    help="Path to the tdp vars",
+)
+@click.option("--dry", is_flag=True, help="Execute dag without running any operation")
+@click.option(
+    "--id",
+    type=int,
+    metavar="dep_id",
+    default=-1,
+    help="Deployment id to resume, defaults to the latest deployment id",
+)
+def resume(
+    database_dsn,
+    collections,
+    run_directory,
+    vars,
+    dry,
+    id,
+):
+    if not vars.exists():
+        raise click.BadParameter(f"{vars} does not exist")
+    dag = Dag(collections)
+    run_directory = run_directory.absolute() if run_directory else None
+
+    ansible_executor = AnsibleExecutor(
+        run_directory=run_directory,
+        dry=dry,
+    )
+    session_class = get_session_class(database_dsn)
+    with session_class() as session:
+        cluster_variables = ClusterVariables.get_cluster_variables(vars)
+        check_services_cleanliness(cluster_variables)
+
+        deployment_runner = DeploymentRunner(
+            collections, ansible_executor, cluster_variables
+        )
+        if id == -1:
+            resumed_deployment_log = get_last_deployment(session_class)
+            click.echo(f"Resuming latest deployment")
+        else:
+            resumed_deployment_log = get_deployment(session_class, id)
+            click.echo(f"Resuming deployment #{id}")
+        targets, sources, filter = ("",) * 3
+        if resumed_deployment_log.targets is not None:
+            targets = resumed_deployment_log.targets
+        if resumed_deployment_log.sources is not None:
+            sources = resumed_deployment_log.sources
+        if resumed_deployment_log.filter_expression is not None:
+            filter = resumed_deployment_log.filter_expression
+        try:
+            deployment_plan_to_resume = DeploymentPlan.from_dag(
+                dag, targets, sources, filter
+            )
+            deployment_plan = DeploymentPlan.from_resume(
+                deployment_plan_to_resume, resumed_deployment_log
+            )
```

Comment:
`from_resume` should only take the failed deployment in input.

`from_resume` is a bit weird because it gives the impression the deployment plan generated is from a resume, you should change to `resume_from` or `from_failed_deployment`

Reply:
I prefer `from_failed_deployment`.

Reply:
Done in the latest commit.

I do believe that we still have to pass `dag` around right ?

Reply:
AH, yes you are right !